plugins/conditionalrestrictions.py

```diff
@@ -0,0 +1,210 @@
+                  {"highway": "residential", "access:forward:conditional": "no @ (10:00-18:00 AND length>5)"},
+                  {"highway": "residential", "access:conditional": "no @ 2099"},
+                  {"highway": "residential", "access:conditional": "no @ (2099 May 22-2099 Oct 7)"},
+                  {"highway": "residential", "access:conditional": "no @ (2010 May 22-2099 Oct 7)"},
+                 ]:
+          assert not a.way(None, t, None), a.way(None, t, None)
+
+        # Expired conditionals
+        for t in [{"highway": "residential", "access:forward:conditional": "no @ 2020"},
+                  {"highway": "residential", "access:conditional": "no @ (2018 May 22-2020 Oct 7)"},
+                  {"highway": "residential", "access:conditional": "no @ (2018 May 22-2020 Oct 7); delivery @ 2099"},
+                  {"highway": "residential", "access:conditional": "no @ (2018 May 22-2020 Oct 7); destination @ length < 4"},
+                  {"highway": "residential", "access:conditional": "no @ (2018 May 22-2020 Oct 7 AND weight > 5)"},
+                 ]:
+          assert a.way(None, t, None), a.way(None, t, None)
+
+        # Invalid conditionals
+        for t in [{"highway": "residential", "access:conditional": "no"},
+                  {"highway": "residential", "access:forward:conditional": "no @ 2098;2099"},
+                  {"highway": "residential", "access:conditional": "delivery @ Mo-Fr 06:00-11:00,17:00-19:00;Sa 03:30-19:00"},
+                  {"highway": "residential", "access:conditional": "delivery @ Mo-Fr 06:00-11:00,17:00-19:00;Sa 03:30-19:00)"},
+                  {"highway": "residential", "access:conditional": "delivery @ (Mo-Fr 06:00-11:00,17:00-19:00;Sa 03:30-19:00"},
+                  {"highway": "residential", "access:conditional": "delivery @ (Mo-Fr 06:00-11:00,17:00-19:00;Sa 03:30-19:00))"},
+                  {"highway": "residential", "access:conditional": "delivery @ Mo-Fr 06:00-11:00,17:00-19:00;Sa 03:30-19:00);yes@wet"},
+                  {"highway": "residential", "access:conditional": "delivery @ (Mo-Fr 06:00-11:00,17:00-19:00;Sa 03:30-19:00;yes@wet"},
+                  {"highway": "residential", "access:conditional": "delivery @ (Mo-Fr 06:00-11:00,17:00-19:00;Sa 03:30-19:00));yes@wet"},
+                  {"highway": "residential", "access:conditional": "yes@()"},
+                  {"highway": "residential", "access:conditional": "yes@"},
+                  {"highway": "residential", "access:conditional": "@wet"},
+                  {"highway": "residential", "access:conditional": "no @ (2099 May 22 AND AND 2099 Oct 7)"},
+                  {"highway": "residential", "access:conditional": "no @ (2099 May 22 AND 2099 Oct 7 AND); delivery @ wet"},
+                  {"highway": "residential", "access:conditional": "no @ (2099 May 22 and 2099 Oct 7); delivery @ wet"},
```

Comment:
I would add one testcase with no space, like: "no@wet".

Reply:
If we'd indeed add a warning for this I'll add a test case to the 'invalid' tests. If we don't, I'll add it to the 'valid' tests.
I'll also update the existing test cases to not trigger on a missing space, as the assert doesn't check which error is triggered, if we add the "missing spaces" warning now or in the future.

Reply:
Resolved for now, check must be moved if the no-space-check must be added in this PR